### PR TITLE
fix(examples): bump gazelle in examples/build_file_generation

### LINKED ISSUE #1372: Gazelle example in `examples/build_file_generation` is broken.
# 🐞 bug report

### Affected Rule

The gazelle examples fail to build on debian and MacOS using rossetta.

### Is this a regression?

 Yes: it worked in `0.20.0` and in `0.23.0`.

### Description

 A clear and concise description of the problem...
The example shared in this repo fails at the third step when autogenerating a python repo using gazelle. 

## 🔬 Minimal Reproduction

Follow the instructions in `examples/build_file_generation` the `bazel run //:gazelle` step generates the error.
## 🔥 Exception or Error

<pre><code>

❯ bazel run //:gazelle
INFO: Analyzed target //:gazelle (38 packages loaded, 247 targets configured).
INFO: Found 1 target...
ERROR: /home/oliver/.cache/bazel/_bazel_oliver/b190e0a64341a6def97c1f54c204cc09/external/rules_python_gazelle_plugin/python/BUILD.bazel:5:11: GoCompilePkg external/rules_python_gazelle_plugin/python/python.a [for tool] failed: (Exit 1): builder failed: error executing command (from target @rules_python_gazelle_plugin//python:python) bazel-out/k8-opt-exec-2B5CBBC6/bin/external/go_sdk/builder_reset/builder compilepkg -sdk external/go_sdk -installsuffix linux_amd64 -src external/rules_python_gazelle_plugin/python/configure.go -src ... (remaining 65 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
external/rules_python_gazelle_plugin/python/lifecycle.go:23:11: undefined: language.BaseLifecycleManager
compilepkg: error running subcommand external/go_sdk/pkg/tool/linux_amd64/compile: exit status 2
Target //:gazelle failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 1.032s, Critical Path: 0.20s
INFO: 9 processes: 9 internal.
FAILED: Build did NOT complete successfully
ERROR: Build failed. Not running target

</code></pre>

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
Debian 10 and MacOS 13.2.1 (22D68) with Rosetta
  </code>
</pre>

**Output of `bazel version`:**

<pre>
  <code>
bazel 6.3.2
  </code>
</pre>

**Rules_python version:**

<pre>
  <code>
0.24.0
  </code>
</pre>

**Anything else relevant?**

### PR DESCRIPTION
**Pull request by OliverFM**
This PR fixes some broken examples in `examples/build_file_generation` – because `gazelle` needed to be updated.

After this PR, `bazel run //:requirements.update && bazel run //:gazelle_python_manifest.update && bazel run //:gazelle` no runs. Moreover if you delete the autogenerated sections of the `BUILD.bazel` file, it will regenerate them.

Fixes #1372

### CODE REVIEW

**Comment by aignas** [comment]
Thanks for the PR @OliverFM, could you please sign the CLA and fix `buildifier` errors?

**Comment by OliverFM** [comment]
> Thanks for the PR @OliverFM, could you please sign the CLA and fix `buildifier` errors?

Will do! I've been away for the last week, but can fix this sometime in the next few days when I am back with my laptop :)

**Comment by OliverFM** [comment]
@aignas I think that this is all sorted – though the CLA change has not showed up yet

**Comment by aignas** [comment]
@OliverFM It seems that `❌ https://github.com/bazelbuild/rules_python/commit/206537498aae89053230087ba987618cc5fb14f9 Author: <ol***r​@wayve.ai>` is the offender with the CLA. Could you please rewrite the git commit history with `git commit --amend --author` please? Squashing everything to a single commit and then doing it should fix the issue.

File: examples/build_file_generation/WORKSPACE [hunk 1]
```diff
@@ -28,13 +28,16 @@ http_archive(
 )
 
 # Download the bazel_gazelle ruleset.
+GAZELLE_VERSION = "0.33.0"
```
Comment (f0rmiga) [inline examples/build_file_generation/WORKSPACE:31]: This is actually a bad refactor, especially to put into examples. Many people use renovate to bump dependencies and it won't understand this. Could you bump the dependency without making this refactor, please?

**Review by f0rmiga — DISMISSED** [review]
Thanks for the PR!

File: examples/build_file_generation/WORKSPACE (hunk 1 above)
Comment (OliverFM) [reply examples/build_file_generation/WORKSPACE:31]: Ah I see, I hadn't realised that was why it was setup that way. I will make that change :)

Comment (OliverFM) [reply examples/build_file_generation/WORKSPACE]: done -- will rerequest review when CI passes

**Review by aignas — APPROVED** [review]

**Pull request merged**